RFCs/FS-1074-default-interface-member-inteop.md

```diff
@@ -0,0 +1,138 @@
+# F# RFC FS-1074 - Default interface member interop
+
+The design suggestion [Default interface member interop](https://github.com/fsharp/fslang-suggestions/issues/679) has been marked "approved in principle".
+This RFC covers the detailed proposal for this suggestion.
+
+# Summary
+
+.NET Core 3.0 will be adding a new runtime concept: default implementations for interface members. We extend the interface implementation mechanism of F# to allow for consuming default implementations. Production of default implementations is not supported.
+
+Given the following C#:
+
+```csharp
+namespace Dims
+{
+    public interface IHaveADefaultMember
+    {
+        int GetNumber() => 12;
+    }
+}
+```
+
+It is possible to consume the default member in F#:
+
+```fsharp
+open Dims
+
+type C() =
+    interface IHaveADefaultMember
+
+let i = C() :> IHaveADefaultMember
+printfn "%d" i.GetNumber() // 12
+```
+
+# Motivation
+
+Because this is a new, fundamental concept added to .NET, F# must understand it.
+
+# Detailed design
+
+## Interface implementation
+
+Interface implementation is extended to not require explicit implementation of members that have a default implementation. That is, the following C# code:
+
+```csharp
+namespace Dims
+{
+    public interface IHaveADefaultMember
+    {
+        int GetNumber() => 12;
+    }
+}
+```
+
+Already has an implementation for the `GetNumber` member. So when a type is specified as implementing `IHaveADefaultMember`, there is no need to specify an implementation of `GetNumber` in the corresponding F# code.
+
+For F# classes, only `interface IHaveADefaultMember` is required:
+
+```fsharp
+open Dims
+
+type C() =
+    interface IHaveADefaultMember
+
+let i = C() :> IHaveADefaultMember
+printfn "%d" i.GetNumber() // 12
+```
+
+For F# object expressions, only `new IHaveADefaultMember` is required:
+
+```fsharp
+open Dims
+
+let i' = { new IHaveADefaultMember }
+printfn "%d" i'.GetNumber() // 12
+```
+
+If the interface being implemented has some members that do not have a default implementation, it is still required to implement them in F# code. Only the members with a default implementation do not require implementing in F# code.
+
+## Overriding a default implementation
+
+It is possible to override a default implementation when implementing an interface in F# code:
+
+```fsharp
+open Dims
+
+type C() =
+    interface IHaveADefaultMember with
+        override __.GetNumber() = 13
```

Comment:
`override` is not what we want. Using `member` is fine. C# has refrained from using `override` word.